smrf/distribute/precipitation.py

```diff
@@ -253,41 +255,43 @@ def distribute(self, data, dpt, precip_temp, ta, time, wind, temp, az,
         """
 
         self._logger.debug('%s Distributing all precip' % data.name)
-        # only need to distribute precip if there is any
         data = data[self.stations]
 
-        if self.nasde_model == 'marks2017':
-            #Adjust the precip for undercatchment
-            if self.config['adjust_for_undercatch']:
-                self._logger.debug('%s Adjusting precip for undercatch...' % data.name)
-                self.corrected_precip.loc[time] = \
-                    precip.adjust_for_undercatch(self.corrected_precip.loc[time],
-                                                 wind,
-                                                 temp,
-                                                 self.config,
-                                                 self.metadata)
-
-            #Use the clipped and corrected precip
-            self.distribute_for_marks2017(self.corrected_precip.loc[time],
-                                          precip_temp,
-                                          ta,
-                                          time,
-                                          mask=mask)
+        if self.config['distribution'] != 'grid':
+            if self.nasde_model == 'marks2017':
```

Comment:
It does actually run the piecewise model if it is requested. Because it is used exactly like susong1999 it is ran in the same distribution func and the model name is passed to cal_phase_and_density which chooses the model function. Marks2017 needed a ton of infrastructure so it got lumped into its own distribution function.